app/src/main/java/org/p2p/wallet/striga/onboarding/StrigaOnboardingPresenter.kt

```diff
@@ -4,42 +4,38 @@ import kotlinx.coroutines.launch
 import org.p2p.wallet.auth.repository.Country
 import org.p2p.wallet.common.mvp.BasePresenter
 import org.p2p.wallet.infrastructure.dispatchers.CoroutineDispatchers
+import org.p2p.wallet.striga.onboarding.StrigaOnboardingContract.View.AvailabilityState
 import org.p2p.wallet.striga.onboarding.interactor.StrigaOnboardingInteractor
 
 class StrigaOnboardingPresenter(
     dispatchers: CoroutineDispatchers,
     private val interactor: StrigaOnboardingInteractor,
 ) : BasePresenter<StrigaOnboardingContract.View>(dispatchers.ui), StrigaOnboardingContract.Presenter {
 
-    override fun attach(view: StrigaOnboardingContract.View) {
-        super.attach(view)
-
+    override fun firstAttach() {
+        super.firstAttach()
         launch {
-            onCountrySelected(interactor.getDefaultCountry())
+            onCurrentCountryChanged(interactor.getChosenCountry())
         }
     }
 
-    override fun onCountrySelected(country: Country) {
-        view?.setCurrentCountry(country)
-        if (isCountrySupported(country)) {
-            view?.setAvailabilityState(StrigaOnboardingContract.View.AvailabilityState.Available)
-
-            // todo: save country
-            // interactor.saveUserCountry(country)
-        } else {
-            view?.setAvailabilityState(StrigaOnboardingContract.View.AvailabilityState.Unavailable)
+    override fun onCurrentCountryChanged(selectedCountry: Country) {
+        launch {
+            interactor.saveCurrentCountry(selectedCountry)
+
+            view?.setCurrentCountry(selectedCountry)
+            if (isCountrySupported(selectedCountry)) {
+                view?.setAvailabilityState(AvailabilityState.Available)
```

Comment:
why we don't use just boolean instead ? x D

Reply:
explicit is better than implicit